New sidebar on essentials page — Preparation to make Tests a key concept on Semaphore 2.0

shiroyasha opened the pull request:
This is the first step in making "Tests" a key concept in our product/documentation.
In this PR, I'm reorganizing the Essentials page to make a proper place under the sun for tests.

The thought process was the following:

**Step 1) Current State in Production**

The list of essentials is too huge. Definitely longer than the average attention span of a reader. 
Also, the list is not ordered in any meaningful way so it is hard to consume.

<img width="1550" alt="Screenshot 2021-09-27 at 18 23 48" src="https://user-images.githubusercontent.com/1779493/134950623-6f05f3b4-81ed-491f-985a-4a7cafc31bea.png">

**Step 2) Essentials organized by topic**

I re-organized the links and grouped them into related categories. The group names emerged after the organization.
Tests are introduced as a key concept in this step. 

<img width="1550" alt="Screenshot 2021-09-27 at 18 23 55" src="https://user-images.githubusercontent.com/1779493/134950848-dca33950-4218-4f34-832b-35fccb6cd0a8.png">

**Step 3) Optimizing the visual aspect of the essentials sidebar**

Nested elements in the navigation were visually smushed together. A stronger separation was necessary. I've added some custom CSS to make it nicer.
 
<img width="1550" alt="Screenshot 2021-09-27 at 18 27 31" src="https://user-images.githubusercontent.com/1779493/134951293-6e76cff1-3988-49b5-ac40-476b79388567.png">

---

The navigation changes kept the links intact. Every page is still under the same URL.

markoa commented:
I like this reorg. IMO job matrix + monorepo belong to Optimizations. Private dependencies is an orphan for a reason, it's not a feature but an example.

shiroyasha commented:
Thank you. I applied this feedback as well, namely:

- Moved Job Matrix and Monorepo to optimizations
- Moved Private dependencies to examples as "Using private dependencies"

This left us with an uncategorized "Debugging with SSH". I moved that to Key Concepts as "Debugging".

End result:
<img width="1504" alt="Screenshot 2021-09-27 at 21 10 27" src="https://user-images.githubusercontent.com/1779493/134970416-67cf6a19-9f3c-491c-bd82-9cf08983ce15.png">
